src/main/java/org/ibp/api/java/impl/middleware/germplasm/GermplasmServiceImpl.java

```diff
@@ -240,14 +254,14 @@ public int searchGermplasmCount(final String searchText) {
 	}
 
 	@Override
-	public GermplasmDTO getGermplasmDTObyGID(final Integer germplasmId) {
-		final Optional<GermplasmDTO> germplasmDTOOptional = this.germplasmService.getGermplasmDTOByGID(germplasmId);
+	public GermplasmDTO getGermplasmDTObyGUID(final String germplasmUUID) {
+		final Optional<GermplasmDTO> germplasmDTOOptional = this.germplasmService.getGermplasmDTOByGUID(germplasmUUID);
```

Comment:
1. Call this.germplasmValidator.validateGermplasmGUID and throw the exception right away. You can create a `validateGUID` method:
```
this.errors = new MapBindingResult(new HashMap<String, String>(), Integer.class.getName());
		this.germplasmValidator.validateGermplasmGUID(errors, germplasmDbId);
		if (this.errors.hasErrors()) {
			throw new ResourceNotFoundException(this.errors.getAllErrors().get(0));
		}
```
2. Remove `isPresent` checking since you've already validated the GUID exists
3. Remove throwing of ApiRuntimeExeption("invalid germplasmDbId)